tasks/create-advisory/create-advisory.yaml

```diff
@@ -72,18 +70,77 @@ spec:
 
         pipelinerun_label="internal-services.appstudio.openshift.io/pipelinerun-uid"
 
+        # only 2 gitlab instances are permitted...prod and staging
+        # these are their secret names
+        #
+        prodSecretName="create-advisory-prod-secret"
+        stagingSecretName="create-advisory-stage-secret"
+        #
+        # detect which one to use based on repositories specified
+        #
+        pending_repositories=$(jq -r '.mapping.components[] | select(.repository | contains("quay.io/redhat-pending/"))\
+          | .repository' $(workspaces.data.path)/$(params.dataPath))
+        prod_repositories=$(jq -r '.mapping.components[] | select(.repository | contains("quay.io/redhat-prod/")) \
+          | .repository' $(workspaces.data.path)/$(params.dataPath))
+        orphan_repositories=$(jq -r '.mapping.components[] | select(.repository | contains("quay.io/redhat-prod/") \
+          | not) | select(.repository | contains("quay.io/redhat-pending/") | not) | \
+          .repository' $(workspaces.data.path)/$(params.dataPath))
+
+        foundPendingRepositories=false
+        if [ ! -z "${pending_repositories[@]}" ]; then
```

Comment:
AFAIK, `! -z` is the same as `-n`, so you it would make sense to use the latter.

Also, the variable was not created as an array, so I don't think it makes sense to use `[@]` here. I did a quick test and it seems that if you try to work with a string as an array, it will just contain one item with everything:
```sh
repos=$(jq -r '.repos[] | select(.repo) | .repo' <<< '{"repos": [{"repo": "myrepo"},{"repo":"myrepo2"}]}')
echo ${repos[@]}
myrepo myrepo2
echo ${repos[0]}
myrepo myrepo2
```

So I would just do this here:
```sh
if [ -n "${pending_repositories}" ]; then
```

The same applies for the others as well.